Track peak memory usage

driazati opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#65157**
* #65155

This adds a test to gather peak memory usage of `import torch` on master and shows an example usage of the `rds_write` APIs

Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

driazati committed:
[wip] Track peak memory usage

[ghstack-poisoned]

driazati committed:
Update on "[wip] Track peak memory usage"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Track peak memory usage"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Track peak memory usage"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Track peak memory usage"




[ghstack-poisoned]

driazati committed:
Update on "[wip] Track peak memory usage"




[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

malfet commented on test/test_import_time.py:
```diff
@@ -7,13 +23,48 @@
 # S3 bucket (see tools/stats/print_test_stats.py)
 class TestImportTime(TestCase):
     def test_time_import_torch(self):
-        TestCase.runWithPytorchAPIUsageStderr('import torch')
+        TestCase.runWithPytorchAPIUsageStderr("import torch")
```
Unrelated to this PR, isn't it? And if it's a new flake8 rule, can we try to enforce it consistently across the codebase?

driazati replied:
Yup, both this and the other PR got hit by format-on-save in VSCode. Maybe we could periodically run it on a allowlist of files in the codebase? Otherwise anyone else with it enabled will hit similar issues (especially since the formatter doesn't support formatting ranges, only entire files AFAIK)

langong347 commented:
Perhaps we can consider changing the filename to `test_import_stats.py` or something like that?

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).

driazati committed:
Update on "Track peak memory usage"


Differential Revision: [D31029049](https://our.internmc.facebook.com/intern/diff/D31029049)

[ghstack-poisoned]

driazati commented:
@driazati has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31029049).
